script.js

```diff
@@ -0,0 +1,368 @@
+    else shiftCaseLeft = true;
+  }
+
+  if (event.code === 'AltLeft') {
+    if (altCaseLeft) altCaseLeft = false;
+    else altCaseLeft = true;
+  }
+  if (event.code === 'CapsLock') {
+    if (capsLock === true) {
+      setStyle(
+        document.querySelector(`.${keyboardElements[event.code][0]}`),
+        false
+      );
+    }
+    capsButtonIsPress();
+  }
+
+  if (event.key === 'shift' || event.key === 'rShift' || event.key === 'Shift') {
+    capsButtonIsPress();
+    return;
+  }
+
+  if (updown === 'down') {
+    if (event.code === 'Tab') {
+      tabCase();
+    }
+
+    if (event.code === 'Backspace') {
+      backspaceButtonISPress();
+    }
+
+    if (event.code === 'Delete') {
+      deleteButtonIsPress();
+    }
+
+    if (shiftCase === false && sessionStorage.lang === 'ENG') {
+      if (keyboardElements[event.code][1] !== undefined) {
+        printKeyCode(keyboardElements[event.code][1]);
+      }
+    }
+
+    if (shiftCase === true && sessionStorage.lang === 'ENG') {
+      if (keyboardElements[event.code][2] !== undefined) {
+        printKeyCode(keyboardElements[event.code][2]);
+      }
+    }
+
+    if (shiftCase === false && sessionStorage.lang === 'RUS') {
+      if (keyboardElements[event.code][3] !== undefined) {
+        printKeyCode(keyboardElements[event.code][3]);
+      }
+    }
+
+    if (shiftCase === true && sessionStorage.lang === 'RUS') {
+      if (keyboardElements[event.code][4] !== undefined) {
+        printKeyCode(keyboardElements[event.code][4]);
+      }
+    }
+  }
+}
+
+function capsButtonIsPress() {
+  if (capsLock === true) capsLock = false;
+  else capsLock = true;
+
+  if (shiftCase === true) shiftCase = false;
+  else shiftCase = true;
+}
+
+function deleteButtonIsPress() {
+  let arr = textarea.value.split('');
+  let position = textarea.selectionStart;
+  arr.splice(position, 1);
+  textarea.value = arr.join('');
+  textarea.selectionStart = textarea.selectionEnd = position;
+}
+
+function backspaceButtonISPress() {
+  let arr = textarea.value.split('');
+  let position = textarea.selectionStart;
+  if (position === 0) return;
+  arr.splice(position - 1, 1);
+  textarea.value = arr.join('');
+  textarea.selectionStart = textarea.selectionEnd = position - 1;
+}
+
+function printKeyCode(key) {
+  let arr = textarea.value.split('');
+  let position = textarea.selectionStart;
+  arr.splice(position, 0, key);
+  textarea.value = arr.join('');
+  textarea.selectionStart = textarea.selectionEnd = position + 1;
+}
+
+function tabCase() {
+  let arr = textarea.value.split('');
+  let position = textarea.selectionStart;
+  arr.splice(position, 0, '    ');
+  textarea.value = arr.join('');
+  textarea.selectionStart = textarea.selectionEnd = position + 4;
+}
+
+function checkLanguage() {
+  if (changeLangUpKeyFlag) {
+    if (altCaseLeft === false && shiftCaseLeft === false) {
+      changeLangUpKeyFlag = false;
+      changeLanguage();
+      showKey();
```

Comment:
For me this function do too many things. At least `showKey` can be removed.

Reply:
Fix

Reply:
I still can see it. Are you sure?